ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParseProxy.java

```diff
@@ -48,14 +48,16 @@ public PubsubMessage apply(PubsubMessage message) {
       final String xpp = attributes.get(Attribute.X_PIPELINE_PROXY);
       String xff = attributes.get(Attribute.X_FORWARDED_FOR);
 
-      // If the configured proxyIpAddress is present, then there's a GCP load balancer IP
+      // If any of the configured proxyIpAddresses is present, then there's a GCP load balancer IP
       // we will need to remove; see
       // https://bugzilla.mozilla.org/show_bug.cgi?id=1729069#c8
-      if (xff != null && xff.contains(proxyIpAddress)) {
-        xff = Arrays.stream(xff.split("\\s*,\\s*")) //
-            .filter(ip -> !proxyIpAddress.equals(ip)) //
-            .collect(Collectors.joining(","));
-        countProxyIpAddress.inc();
+      for (String proxyIpAddress : proxyIpAddressCsv.split(",")) {
+        if (xff != null && xff.contains(proxyIpAddress)) {
+          xff = Arrays.stream(xff.split("\\s*,\\s*")) //
```

Comment:
I see elsewhere in this repo (ctxsvc-reporter job) the pattern of `.filter(StringUtils::isNotBlank)` instead of splitting with spaces as is done here. Perhaps this should be treated the same way.

Reply:
This particular line is unchanged from previous implementation (just it's now nested in a for loop). If we ended up with a blank entry here, I think it would mean that the input `xff` included a trailing comma, which would be very unexpected and which might cause additional debugging confusion if we filter it out implicitly here.

Reply:
But it would be good to to add `filter(StringUtils::isNotBlank)` when we do `proxyIpAddressCsv.split(",")` above.